### ISSUE TITLE
FR: log error if Kubernetes secret fetch fails

### ISSUE THREAD

**Issue by garrett-ts** [tailscale/tailscale#6039]
### What are you trying to do?

From a support ticket - 6206.

We should add better error messaging for Kubernetes support. 

One example that makes troubleshooting difficult is when sidecar fails to acquire the secret (because of permissions or misconfiguration) it doesn't throw an error or even emit log messages.

**Comment by DentonGentry** [comment]
We can address a specific item in the feedback: "when sidecar fails to acquire the secret (because of permissions or misconfiguration) it doesn't throw an error or even emit log messages"

**Comment by maisem** [comment]
This seems to have been fixed with #6646 and #6649

I now see
```
➜  k8s git:(main) ✗ k logs nginx ts-sidecar
boot: 2022/12/16 14:02:09 Some Kubernetes permissions are missing, please check your RBAC configuration: multiple errors:
	missing get permission on secret "tailscale"
	missing update permission on secret "tailscale"
```
